spaghetti/util.py

```diff
@@ -807,17 +802,9 @@ def _routes_as_gdf(paths, id_col, geom_col):
 
     """
 
-    # isolate the origins, destinations, and geometries
-    origs = [o for (o, d), g in paths]
-    dests = [d for (o, d), g in paths]
-    geoms = [LineString(g.vertices) for (o, d), g in paths]
-
     # instantiate as a geodataframe
-    paths = geopandas.GeoDataFrame(geometry=geoms)
-    paths["O"] = origs
-    paths["D"] = dests
-
-    if id_col:
-        paths[id_col] = paths.apply(lambda x: (x["O"], x["D"]), axis=1)
+    paths = dict(paths)
+    ids, geoms = zip(paths.keys()), [LineString(g.vertices) for g in paths.values()]
```

Comment:
Can we avoid this scalar constructor?

Reply:
> Can we avoid this scalar constructor?

Not sure what you mean here. Can you please elaborate?

Reply:
I meant if we can create all LineStrings at once using `shapely.linestrings` instead of calling the scalar `LineString` constructor in a loop.

Reply:
I think we are unable to do that because routes are made of two or more vertices, where all routes passed into the function can have a different number of vertices. It seems as though `shapely.linestrings()` needs to have all inputs as the same number of vertices for a non-ragged array. So using `shapely.linestrings()` works in `_arcs_as_gdf()` since all input are exactly 2 points, but not in `_routes_as_gdf()`.